core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/service/asset/AssetServiceImpl.java

```diff
@@ -109,4 +109,32 @@ public ServiceResult<Asset> delete(String assetId) {
             return ServiceResult.success(deleted);
         });
     }
+
+    @Override
+    public ServiceResult<Void> update(String assetId, Asset asset) {
+        return transactionContext.execute(() -> {
+            if (findById(assetId) == null) {
```

Comment:
I would add a validation here to ensure that the `assetId` parameter is equal as `asset.getId()`, to avoid the possibility to update the id.

Reply:
that validation will likely fail, because the `AssetUpdateDto` doesn't (on purpose!) have an `id`, so it will be transformed into an `Asset` that does not have an `id` either. That's the reason I tunneled the `assetId` in separately. To me, having the ID separate makes it more explicit.

Reply:
ok, but looking at the `Asset.Builder`:
```
 @Override
        public Asset build() {
            if (entity.getId() == null) {
                id(UUID.randomUUID().toString());
            }
            return super.build();
        }
```

so this Asset will have a randomly generated id, also if that doesn't count because it won't be updated I think it would be better to have it valued correctly and then checked anyway because the service shouldn't rely on controller and store behavior (e.g. in a document db I would expect just a "set the whole object") behavior, and if the id is not correct that will be an issue.

Reply:
ok, so we either need a way to set the `id` after the transformer has created the asset, e.g. with a `Asset.toBuilder` method, or by not using the transformer at all, and "manually" create the asset object.